modules/swagger-parser/src/main/java/io/swagger/parser/processors/ExternalRefProcessor.java

```diff
@@ -208,8 +208,17 @@ private void processDiscriminator(String discriminator, Map<String, Property> pr
                             processRefProperty(new RefProperty(RefType.DEFINITION.getInternalPrefix()+name), file);
                         }
                     }
-
-
+                }else if (prop.getValue() instanceof RefProperty) {
+                    String ref = ((RefProperty) prop.getValue()).getSimpleRef();
+                    Object resolved = swagger.getDefinitions().get(ref);
```

Comment:
does this mean that the ref must have been already resolved? are we sure it will be so?